### PR TITLE
fix: dataset input_shape

### PR DESCRIPTION
**Pull request by streethagore**
Fix dataset input_shape by removing the flattening operation during data-augmentation and replaces it by a flattening operation at the beginning of the forward and extended_forward of GrowingMLP.

### CODE REVIEW

File: src/gromo/growing_mlp.py [hunk 1]
```diff
@@ -25,13 +25,16 @@ def __init__(
             raise NotImplementedError("Device selection is not implemented yet")
         self.device = global_device()
         self.input_shape = input_shape
+        self.num_features = torch.tensor(input_shape).prod().int().item()
```
Comment (TheoRudkiewicz) [inline src/gromo/growing_mlp.py:28]: I am sure that e need this to be a class variable and not simply a local variable (but it's not an important remark).

**Review by TheoRudkiewicz — COMMENTED** [review]
Yes thank you for correcting this potential bug in dataset ! In addition I was wondering if you could not correct other potential bugs at the same time. Indeed for now we transform data to tensors and normalize it before the data augmentation. For rotations and flips I don't see this as a problem. However for color augmentation I don't know if it should be done before or after normalization. If it is before, we should correct the function now.

**Review by sylvchev — APPROVED** [review]
Thanks for catching this. For color augmentation, I let you decide if you want to update this PR or open a new one.

**Comment by sylvchev** [comment]
@streethagore  You could update the docs/source/whatsnew.yml to add this bugfix

File: src/gromo/growing_mlp.py (hunk 1 above)
Comment (sylvchev) [reply src/gromo/growing_mlp.py:28]: For batch norm?
@TheoRudkiewicz as you said it was not import, I suggest we merge this code.

**Review by sylvchev — APPROVED** [review]
